### PR TITLE
Get transformer rejection test working; test transformer handling of …

### PR DESCRIPTION
**Pull request by bstansberry**
…legacy-empty-beans-xml-treatment=true

@manovotn This adds what the commit msg says to you EE1_aaditions branch.

I'm going to send up a draft PR to wildfly of this branch as well, but that's just to let CI have at it; I'll close that tomorrow.

### CODE REVIEW

File: weld/subsystem/src/main/java/org/jboss/as/weld/WeldTransformers.java [hunk 1]
```diff
@@ -27,9 +28,9 @@ public void registerTransformers(SubsystemTransformerRegistration subsystem) {
         ResourceTransformationDescriptionBuilder builder400 = chainedBuilder.createBuilder(subsystem.getCurrentSubsystemVersion(), version4_0_0);
         builder400.getAttributeBuilder()
                 // Reject if false or undefined, discard otherwise
-                .addRejectCheck(RejectAttributeChecker.UNDEFINED, WeldResourceDefinition.LEGACY_EMPTY_BEANS_XML_TREATMENT_ATTRIBUTE)
-                .addRejectCheck(new RejectAttributeChecker.SimpleAcceptAttributeChecker(WeldResourceDefinition.LEGACY_EMPTY_BEANS_XML_TREATMENT_ATTRIBUTE.getDefaultValue()), WeldResourceDefinition.LEGACY_EMPTY_BEANS_XML_TREATMENT_ATTRIBUTE)
-                .setDiscard(DiscardAttributeChecker.ALWAYS, WeldResourceDefinition.LEGACY_EMPTY_BEANS_XML_TREATMENT_ATTRIBUTE)
+                .setDiscard(new DiscardAttributeChecker.DiscardAttributeValueChecker(false, true, ModelNode.TRUE),
```
Comment (manovotn) [inline weld/subsystem/src/main/java/org/jboss/as/weld/WeldTransformers.java:31]: Maybe I'm misreading this but doesn't this discard the config if its undefined as well? I thought we wanted to reject it in such case.

**Review by manovotn — APPROVED** [review]
Thanks a lot @bstansberry.
I'll merge this and adapt the original PR to have links to JIRA issues.

File: weld/subsystem/src/main/java/org/jboss/as/weld/WeldTransformers.java (hunk 1 above)
Comment (bstansberry) [reply weld/subsystem/src/main/java/org/jboss/as/weld/WeldTransformers.java:31]: Oh, crud; you are right! I mixed up the params. I merged your branch; I'll send up a PR to fix this.

Comment (manovotn) [reply weld/subsystem/src/main/java/org/jboss/as/weld/WeldTransformers.java:31]: Cool, thanks!

**Pull request merged**